ops/cli.py

```diff
@@ -1,351 +1,309 @@
-        pass
-    return f"\033[1;{color_code}m{message}\033[0m"
-
-
-def info(message: str):
-    """Log the info to stdout"""
-    print(_apply_color("default", message))
-
-
-def success(message: str):
-    """Log the success to stdout"""
-    print(_apply_color("green", message))
-
-
-def error(message: str):
-    """Log the error to stderr"""
-    print(_apply_color("red", message), file=sys.stderr)
-
-
-def warn(message: str):
-    """Log the warning to stdout"""
-    print(_apply_color("yellow", message))
-
-
-def registerableCLI(cls):
-    """Class decorator to register methodss with @register into a set."""
-    cls.registered_commands = set([])
-    for name in dir(cls):
-        method = getattr(cls, name)
-        if hasattr(method, "registered"):
-            cls.registered_commands.add(name)
-    return cls
-
-
-def register(func: Callable) -> Callable:
-    """Method decorator to register CLI commands."""
-    func.registered = True
-    return func
-
 
-def shell(command: str) -> str:
-    """Run COMMAND in a subprocess."""
-    info(f"Running: {command}")
-    return subprocess.check_output(command, shell=True, encoding="utf-8").strip()
-
-
-def shell_unchecked(command: str):
-    """Run COMMAND in a subprocess and who cares whether it fails!"""
-    warn(f"Running unchecked: {command}")
-    return subprocess.call(command, shell=True)
-
-
-def clone(url: str, branch: str):
-    """Clone the BRANCH of git repo at URL."""
-    info(f"=> Cloning {branch} of {url} ...")
-    shell(f"git clone {url} --branch {branch}")
-    success(f"[✔] Cloned {branch} of {url}.")
-
-
-def render_ctmpl(ctmpl_file: str, **kwargs) -> int:
-    """Render a ctmpl file."""
-    info("=> Rendering ctmpl file {ctmpl_file}")
-    envs = ""
-    if kwargs:
-        for k, v in kwargs.items():
-            envs += f"-e {k}={v}"
-    info(f"=> Feeding variables: {envs}")
-    command = " ".join(['docker run -i --rm -v "$(pwd)":/working',
-                        '-v "$HOME"/.vault-token:/root/.vault-token',
-                        f'{envs}',
-                        'broadinstitute/dsde-toolbox:dev',
-                        '/usr/local/bin/render-ctmpls.sh -k',
-                        f'"{ctmpl_file}"'])
-    shell_unchecked(command)
-    success(f"[✔] Rendered file {ctmpl_file.split('.ctmpl')[0]}")
-
-
-def commands_available():
-    commands = ["docker", "gcloud", "git", "helm", "java", "kubectl", "jq"]
-    for cmd in commands:
-        if not shutil.which(cmd):
-            error(f"=> {cmd} is not in PATH. Please install it!")
-
-
-def set_up_helm():
-    info("=> Setting up Helm charts")
-    ADD = " ".join(["helm repo add gotc-charts",
-                    "https://broadinstitute.github.io/gotc-helm-repo/"])
-    shell(ADD)
+@dataclass
+class WflInstanceConfig:
+    """Data class representing the 'state' of this script's configuration as it interacts with infrastructure."""
+    command: str
+    environment: str
+    instance_id: str
```

Comment:
Without a default these become required by `@dataclass` for construction (but the CLI parsing also enforces that these are required anyway)